youtube_dl/extractor/francetv.py

```diff
@@ -191,66 +217,260 @@ def sign(manifest_url, manifest_id):
+
+            # avoid duplicating title text
+            titles = ('title', 'additional_title')
+            title_parts = [
+                re.sub(r'(?:_|[^\w])+', ' ', t).strip()
+                for t in traverse_obj(dinfo, ('meta', titles))]
+            if len(title_parts) == 2 and title_parts[0] == title_parts[1]:
+                titles = titles[:1]
+
+            info = merge_dicts(info, traverse_obj(dinfo, ('meta', {
+                'title': T(lambda d: join_nonempty(
+                    *((d.get(t) or '').strip() or None
+                      for t in titles), delim=' - ')),
+                'description': ('description', T(lambda s: s.strip() or None)),
+                'thumbnail': ('image_url', T(url_or_none)),
+                'timestamp': ('broadcasted_at', T(parse_iso8601)),
+                'duration': ('playtime', T(parse_duration)),
+                'alt_title': 'additional_title',
+            })), traverse_obj(dinfo, (
+                # meta['pre_title'] contains season and episode number for series in format "S<ID> E<ID>"
+                'meta', 'pre_title', T(lambda x: re.search(
+                    r'S(\d+)\s*E(\d+)', x)), {
+                        'season_number': (1, T(int_or_none)),
+                        'episode_number': (2, T(int_or_none)),
+                })))
+
+        if not videos and drm_formats:
+            self.report_drm(video_id)
+
+        formats, subtitles = self._extract_formats_and_subtitles(videos, video_id)
+
+        if info.get('is_live'):
+            info['title'] = self._live_title(info['title'])
+
+        return merge_dicts(info, {
             'formats': formats,
             'subtitles': subtitles,
-        }
+            # '_format_sort_fields': ('res', 'tbr', 'proto'),  # prioritize m3u8 over dash
+        }, {
+            'episode': info.get('alt_title'),
+            'series': info['title'],
+        } if info.get('episode_number') else {})
 
     def _real_extract(self, url):
-        mobj = re.match(self._VALID_URL, url)
-        video_id = mobj.group('id')
-        catalog = mobj.group('catalog')
+        url, smuggled_data = unsmuggle_url(url, {})
+        video_id = self._match_id(url)
+        hostname = smuggled_data.get('hostname') or 'www.france.tv'
 
-        if not video_id:
-            qs = compat_urlparse.parse_qs(compat_urlparse.urlparse(url).query)
-            video_id = qs.get('idDiffusion', [None])[0]
-            catalog = qs.get('catalogue', [None])[0]
-            if not video_id:
-                raise ExtractorError('Invalid URL', expected=True)
+        return self._extract_video(video_id, hostname=hostname)
+
+
+class FranceTVEmbedIE(FranceTVBaseIE):
+    _VALID_URL = r'''(?x)
+        https?://embed\.francetv\.fr(?:/?\?(?:.*&)?(?P<ue>ue)=|/)
+        # Say (?:...|) instead of (?:...)? when ... ends .* to avoid
+        # python/cpython#62847 (fixed from at least 3.5 and late 2.7)
+        (?P<id>[\da-f]{32})(?:(?(ue)&|/?[?#]).*|)$```
```

Comment:
```suggestion
        # Say (?:|...) instead of (?:...)? when ... ends .* to avoid
        # python/cpython#62847 (fixed from at least 3.5 and late 2.7)
        (?P<id>[\da-f]{32})(?:|(?(ue)&|/?[?#]).*)$
```